hello/reddit.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+import json
+import io
+import requests
+from hello.models import Message
+from django.db.utils import IntegrityError
+import datetime
+
+REDDIT_API_ENDPOINT = "https://api.reddit.com/r/investing/search?q={}&type=link&restrict_sr=on"
+
+
+def get_companies():
+    # XXX Search the stocks database in the future
+    with io.open("stocktalk/static/raw_data/symbols_usa_only.json") as f:
+        j = f.read()
+    list_of_companies = json.loads(j)
+    dict_of_companies = {}
+    for company in list_of_companies:
+        dict_of_companies[company['Ticker']] = company
+    return dict_of_companies
+
+
+def ticker_to_name(data, ticker):
+    try:
+        ticker = str(ticker.decode())
+    except AttributeError:
+        pass
```

Comment:
Django utils comes with a couple of functions for ensuring that the thing you have is either bytes or unicode.  You might try using those directly here.  They are pretty rigorously tested, and cover a lot of edge cases you may not have considered.